docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+| During parsing. | No memory resource wasted as the command will not fail due to invalid file. | Decreases cohesion as `Parser` now has to handle file matters on top of parsing matters. |
+
+Having considered both of the alternatives, we have decided to implement the first alternative, **read script file content during command execution** because we do not want to decrease the cohesion of Parser, and we find that the memory resource wasted in the process is a worthy exchange for the cohesion preserved.
+
+<br/>
+
+Aspect: **The way user can specify the script file**
+
+| Approach | Pros | Cons |
+| --- | --- | --- |
+| Specify file extension **(current design)**. |  Ensures the correct file will be read. | Some users may not know how to identify the file extension. |
+| Do not have to specify file extension. | Users can easily specify the file to read. | May read the wrong file due to identical names but different file extension. |
+
+We have decided to the implement the first alternative, **users should specify the file extension in their input** because there is great importance in getting a correct estimation timing, and it far outweighs and compensates for the hassle of entering the file extension, and we believe such mistakes are costly for our users.
+
+<br/>
+
+### 4.2 Browse Feature
+The browse feature is a useful feature that will allow users to quickly look through all 
+the different anime series available in a browsing fashion. The feature will have several enhancements such 
+as sorted browsing to browse in alphabetical or by the rating of the anime.
+
+<br/>
+
+#### 4.2.1 Current Implementation
+The `BrowseCommand` is executed by `BrowseCommandParser` after parsing the user input. It will then fetch `Anime` objects matching the parameters specified by `BrowseCommandParser` that are stored in `AnimeData`. It extends the `Command` class and implements the following operations:
+*   `BrowseCommand#sortBrowseList()` - Handles any sorting of `Anime` objects. 
+*   `BrowseCommand#buildBrowseOutput()` - Creates the output string to be printed to users.
+<br/>
+
+These are the two fundamental operations that will carry out the execution of the browse command.
+
+`BrowseCommand` will be constructed with default values. This ensures that even when there is no input provided, 
+it can still perform a default browse and return relevant useful information.
+
+Before going any further, please refer to this helpful table of the `BrowseCommand` parameters for reference.
+
+| Attribute | Option | Function          |
+| ---       | ---    | ---               |
+| order     | 0      | Ascending         |
+| order	    | 1      | Descending        |
+| sortType  | 0      | No Sort           |
+| sortType  | 1      | by name           |
+| sortType  | 2      | by rating         |
+| sortType  | 3      | back to original  |
+| page      | \>= 1  | page number       |
```

Comment:
I like this concept of using a table as information is presented in a very readable and simple manner. The only question I have is with regards to the heading. Maybe it would be better if you explained to the user what the headings were before introducing them? If not it might lead to further confusion.